React router lesson: Make sentences clearer (#26051)
TheOdinProject/curriculum: react/the_react_ecosystem/react_router.md

### A Reactive solution

> While client-side routing allows for nicer, app-like interactions (since you are controlling the routes, you can make fancy CSS animations across route changes), a lot of caveats can be missed. Browsers reload notify screen-readers of new content to read, so you will need to notify screen-readers of route updates manually. However, with the help of a robust library, you can often address these concerns!

Reviewer: I think this is more likely to be read as "Browser reloads", "When a browser reloads, it notifies screen-readers...", or "A browser reload will notify screen-readers..." than "Browsers reload"

```suggestion
While client-side routing allows for nicer, app-like interactions (since you are controlling the routes, you can make fancy CSS animations across route changes), a lot of caveats can be missed. When a browser reloads, it notifies screen-readers of new content to read, but in the case of client-side routing, you will need to notify screen-readers of route updates manually. However, with the help of a robust library, you can often address these concerns!
```

);
~~~

Much nicer! 

### Protected routes and navigation

> Often, you will need to decide whether a certain route should be rendered or not. One example is authentication, where you render certain routes based on if the user is logged in or not. If they are logged in, you show some information about the user like [here at the dashboard](https://www.theodinproject.com/dashboard). While there are many ways to do so, one of the easiest ways is to conditionally create a config for the router.

Reviewer: I think this will read better and flow better to the next paragraph if we add an "Otherwise" after mentioning "If they are logged in". 

```suggestion
Often, you will need to decide whether a certain route should be rendered or not. One example is authentication, where you render certain routes based on if the user is logged in or not. If they are logged in, you show some information about the user like [here at the dashboard](https://www.theodinproject.com/dashboard). Otherwise, they are redirected to the sign-in page. While there are many ways to do so, one of the easiest ways is to conditionally create a config for the router.
```

> Often, you will need to decide whether a certain route should be rendered or not. One example is authentication, where you render certain routes based on if the user is logged in or not. If they are logged in, you show some information about the user like [here at the dashboard](https://www.theodinproject.com/dashboard). Otherwise, they are redirected to the sign-in page (this could be any page). While there are many ways to do so, one of the easiest ways is to conditionally create a config for the router.

Reviewer 2: ```suggestion
Often, you will need to decide whether a certain route should be rendered or not. One example is authentication, where you render certain routes based on if the user is logged in or not. If they are logged in, you show some information about the user like [here at The Odin Project dashboard page](https://www.theodinproject.com/dashboard). Otherwise, they are redirected to the sign-in page (this could be any page). While there are many ways to do so, one of the easiest ways is to conditionally create a config for the router.
```

- while we are here I thought it would be nice to make this link more descriptive and thus accessible :)

> You will often come across the need to reroute the user to a different URL programmatically. This is where we use the [`<Navigate />`](https://reactrouter.com/en/main/components/navigate) component. `<Navigate />` component reroutes the user to the desired URL when it is rendered. The component itself is a wrapper around [the useNavigate hook](https://reactrouter.com/en/main/hooks/use-navigate) that lets you navigate programmatically, to URLs, or even go back down the user's history.

Reviewer: IMO, we don't need to repeat the name of this component.
```suggestion
You will often come across the need to reroute the user to a different URL programmatically. This is where we use the [`<Navigate />`](https://reactrouter.com/en/main/components/navigate) component. The component reroutes the user to the desired URL when it is rendered. The component itself is a wrapper around [the useNavigate hook](https://reactrouter.com/en/main/hooks/use-navigate) that lets you navigate programmatically, to URLs, or even go back down the user's history.
```

Reviewer: Sounds good @Eveieve :+1:

> You will often come across the need to reroute the user to a different URL programmatically. This is where we use the [`<Navigate />`](https://reactrouter.com/en/main/components/navigate) component. The `<Navigate />` component reroutes the user to the desired URL when it is rendered. It is a wrapper around [the useNavigate hook](https://reactrouter.com/en/main/hooks/use-navigate) that lets you navigate programmatically, to URLs, or even go back down the user's history.

Reviewer 2: ```suggestion
You will often come across the need to reroute the user to a different URL programmatically. This is where we use [the `<Navigate />`component](https://reactrouter.com/en/main/components/navigate). The `<Navigate />` component reroutes the user to the desired URL when it is rendered. It is a wrapper around [the useNavigate hook](https://reactrouter.com/en/main/hooks/use-navigate) that lets you navigate programmatically, to URLs, or even go back down the user's history.
```

- caught another while reviewing. 

Note: if you feel these links (and the others in the lesson) are required reading, they shouldn't be scattered throughout the lesson but placed in the assignment per the style guide. 
If so, that should probably be a separate issue :)
